lib/Bindings/Python/circt/dialects/_comb_ops_ext.py

```diff
@@ -47,7 +49,16 @@ def BinaryOp(base):
   class _Class(base):
 
     @classmethod
-    def create(cls, result_type, **kwargs):
+    def create(cls, result_type=None, **kwargs):
+      if not result_type:
+        types = list(map(lambda arg: arg.type, kwargs.values()))
+        if not types:
```

Comment:
I'm assuming this handles `len(types) == 0`?

Reply:
yep